rescheduler.go

```diff
@@ -141,60 +136,17 @@ func main() {
 
 	recorder := createEventRecorder(kubeClient)
 
-	// Allows active/standy HA.
```

Comment:
What happen if we have 2 pods ? are we going to have concurrent drain ?

Reply:
It wouldn't be ideal, but probably not a big deal given the timings and grace delays. Few interesting tidbits on this:

The way to do leader election changed drastically between that very old version, and the current version. Most stuff either moved or get some mild updates, like background, or the new snapshot concept, but this leader election is just an entire different concept. We are currently not really using this feature at all, it didn't seem wise to spend too much time on this. I'd really be happy to have it back, I can give this some more time, but probably not as part of this PR.

The time where this would likely be most vulnerable is during node rotations, where this component doing a job very similar to the cluster-autoscaler, which is to take automated decisions on nodes that are likely to go against what the node rotation is meant to force, should just be scaled down during those exercises. If it isn't scaled down, there are checks in place to not do any action while there are pending pods, which does happen during node rotations. During drain events it also checks and adds taints to nodes, so two processes running at once would likely catch that as well, barring some pretty crazy conditions. And on top of all that, we have control over the deployment size and can use rollingUpdate strategies to terminate first rather than scale up then terminate. 

Given the scope of the change, my lack of experience with it, the lack of usage of it to begin with and the mitigations available to us, I'm happy with this in this PR I think.

Reply:
If it's a concern during rotation we could downscale it to 0 like we do with cluster-autoscaler